plugins/builder/osbuild.py

```diff
@@ -184,13 +186,47 @@ def as_dict(self):
         return res
 
 
-class ImageStatus(enum.Enum):
-    SUCCESS = "success"
-    FAILURE = "failure"
-    PENDING = "pending"
-    BUILDING = "building"
-    UPLOADING = "uploading"
-    REGISTERING = 'registering'
+class ComposeStatusError:
+    def __init__(self, error_id: int, reason: str, details: Union[Dict, None]):
+        self.error_id = error_id
+        self.reason = reason
+        self.details = details
+
+    def as_dict(self):
+        data = {
+            "id": self.error_id,
+            "reason": self.reason
+        }
+        if self.details:
+            data["details"] = self.details
+        return data
+
+
+class ImageStatus:
```

Comment:
The introduction of this class changes metadata that we save in Brew, right? In that case, it should be a separate commit.

Reply:
Can you be more specific about what you mean by metadata? The build task has some associated metadata which are stored with the build (like the content generator that was used, its version, buildroot, etc). The status is uploaded as a JSON file to the build task. Nevertheless I can extract it into a separate commit.

Reply:
Yes, I meant the JSON file that's uploaded. It's now different, isn't it?

Reply:
done

Reply:
> Yes, I meant the JSON file that's uploaded. It's now different, isn't it?

yes, it is...